Add support for glob patterns in net input plugin

bobmshannon opened the pull request:
An approach for fixing #1073.

Adds support for specifying [golang glob](https://github.com/gobwas/glob)-style patterns.

**Example**
```
[[inputs.net]]
   interfaces = ["en*", "utun[0-1]", "lo0"]
```

```
$ ./telegraf --input-filter net --config telegraf.conf --test
* Plugin: inputs.net, Collection 1
> net,interface=lo0,host=mac bytes_sent=103240327i,packets_sent=227866i,drop_in=0i,drop_out=0i,bytes_recv=103240327i,packets_recv=227866i,err_in=0i,err_out=0i 1503126134000000000
> net,interface=en0,host=macbook packets_sent=13568823i,packets_recv=23715254i,err_in=0i,err_out=0i,drop_out=3193i,bytes_sent=2411560838i,bytes_recv=30527805468i,drop_in=0i 1503126134000000000
> net,interface=en1,host=macbook bytes_sent=0i,drop_out=0i,bytes_recv=0i,packets_sent=0i,packets_recv=0i,err_in=0i,err_out=0i,drop_in=0i 1503126134000000000
> net,interface=en2,host=macbook bytes_recv=0i,packets_sent=0i,packets_recv=0i,err_in=0i,err_out=0i,drop_in=0i,bytes_sent=0i,drop_out=0i 1503126134000000000
> net,interface=utun0,host=macbook bytes_sent=268i,packets_sent=3i,err_in=0i,drop_in=0i,drop_out=0i,bytes_recv=0i,packets_recv=0i,err_out=0i 1503126134000000000
> net,interface=utun1,host=macbook bytes_sent=22156i,packets_recv=115i,drop_in=0i,drop_out=0i,bytes_recv=20275i,packets_sent=108i,err_in=0i,err_out=0i 1503126134000000000
```

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

ljagiello commented:
@danielnelson any chance this PR will be merge before 1.5.0?

danielnelson commented on plugins/inputs/system/net.go:
```diff
@@ -5,12 +5,14 @@ import (
 
 type NetIOStats struct {
-	ps PS
+	patterns []glob.Glob
```
Use filter.Filter here

danielnelson commented:
I think we can get this in, looking over the code I assume the code causing the performance issue is caused by calling `net.InterfaceByName(io.Name)`, which is only called if the `interfaces` option is empty.  So by adding the ability to use globs we bypass that code?  If either of you are experiencing the performance issue can you verify my assumption?

bobmshannon commented:
I think the performance problem can currently be mitigated by explicitly specifying a list of interfaces, it's just that support for glob patterns makes doing so less verbose and more adapting to different environments (e.g. `eth*` vs `eth0,eth1,eth2,eth3,...`)

ljagiello commented:
@bobmshannon It would be super cool if you could add blacklist, I would love to blacklist all calico interfaces.

bobmshannon commented:
@danielnelson I have addressed the PR comments, but it looks like the build may have flaked.

@ljagiello A blacklist sounds like an interesting idea. I'm not sure I can get that in this PR but it may be worth opening a separate issue for. @danielnelson, what do you think about such an option?

danielnelson commented:
We can add the blacklist, usually we have a pair of options like `interface_include` `interface_exclude`.  We could add this in a backwards compatible way by appending the `interfaces` option to `interface_include`.

danielnelson commented:
I think the tests are failing because the branch is so far behind master.  Failures are in unrelated code though so I don't mind merging.
